pkg/agent/route/route_linux.go

```diff
@@ -1595,6 +1595,67 @@ func (c *Client) deleteNodeIP(podCIDR *net.IPNet) error {
 	return nil
 }
 
+func (c *Client) AddIPRule(mark int, table int) error {
+	exist, err := c.checkIfRuleExist(mark, table)
+	if err != nil {
+		return fmt.Errorf("failed to add IP rule on the host, error: %v", err)
+	}
+
+	if !exist {
+		rule := netlink.NewRule()
+		rule.Table = table
+		rule.Mark = mark
+		return c.netlink.RuleAdd(rule)
+	}
+
+	return nil
+}
+
+func (c *Client) DeleteIPRule(mark int, table int) error {
+	exist, err := c.checkIfRuleExist(mark, table)
```

Comment:
Can we know if rule exists or not from the netlink.RuleDel() return error?

Reply:
Yes, using strings.Contains to check now.